ipatests/test_integration/test_nfs.py

```diff
@@ -236,22 +233,61 @@ def test_automount_location(self):
         # maybe re-use m1.group(0) if it exists.
         assert m1 is None
 
+        # https://pagure.io/freeipa/issue/7918
+        # check whether idmapd.conf was setup using the IPA domain
+        automntclt.run_command([
+            "grep", "Domain = %s" % self.master.domain.name, "/etc/idmapd.conf"
+        ])
+
         time.sleep(WAIT_AFTER_INSTALL)
 
         automntclt.run_command([
             "mount", "-t", "nfs4", "-o", "sec=krb5p,vers=4.0",
             "%s:/exports/home" % nfssrv.hostname, "/home", "-v"
         ])
 
+        # TODO leverage users
+
         automntclt.run_command(["umount", "-a", "-t", "nfs4"])
 
         result2 = automntclt.run_command([
-            'ipa-client-automount', '--uninstall',
-            '-U', '--debug'
+            'ipa-client-automount', '--uninstall', '-U', '--debug'
         ])
-
         m2 = re.search(r'(?<=stderr\=Failed).+', result2.stderr_text)
         assert m2 is None
 
         time.sleep(WAIT_AFTER_UNINSTALL)
+
+        # https://pagure.io/freeipa/issue/7918
+        # test for --idmap-domain DNS
+        automntclt.run_command([
+            'ipa-client-automount', '--location', 'default',
+            '-U', '--debug', "--idmap-domain", "DNS"
+        ])
+        # check whether idmapd.conf was setup properly:
+        # grep must not find any configured Domain.
+        result = automntclt.run_command(
+            ["grep", "^Domain =", "/etc/idmapd.conf"], raiseonerr=False
+        )
+        assert result.returncode == 1
+
+        automntclt.run_command([
+            'ipa-client-automount', '--uninstall', '-U', '--debug'
+        ])
+
+        # https://pagure.io/freeipa/issue/7918
+        # test for --idmap-domain exampledomain.net
+        nfs_domain = "exampledomain.net"
+        automntclt.run_command([
+            'ipa-client-automount', '--location', 'default',
+            '-U', '--debug', "--idmap-domain", nfs_domain
+        ])
+        # check whether idmapd.conf was setup using nfs_domain
+        automntclt.run_command([
+            "grep", "Domain = %s" % nfs_domain, "/etc/idmapd.conf"
+        ])
+
```

Comment:
Similar here, on failure will it leave things installed?

Reply:
It should and it's probably better for investigation I guess. Would you like anything changed?

Reply:
It's just that most tests clean themselves up even on failures.